src/CLIFramework/Extension/ExtensionBase.php

```diff
@@ -0,0 +1,42 @@
+<?php
+namespace CLIFramework\Extension;
+
+use CLIFramework\ServiceContainer;
+use CLIFramework\Command;
+use CLIFramework\Hook\Hookable;
+use CLIFramework\Hook\HookHolder;
+use CLIFramework\Exception\ExtensionException;
+
+abstract class ExtensionBase
+    implements Extension, Hookable
+{
+    private $hooks;
+
+    private function getHooks()
```

Comment:
> Please note the event trigger could be application-wide, the event handling should enable us triggering events from everywhere of Application.

I agree with your thought and at first I was going to add an event system globally available. A common implementation is like below.

```php
class Thing
{
    public function do()
    {
        EventBus::subscribe('some.event', function() {
            // handle an event
        });
    }
}

EventBus::notify('some.event');
```